### PR TITLE
feat(manager-components): add base template

### PR DESCRIPTION
**Pull request by aboungnaseng-ovhcloud**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | #14377
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Add new template listing on manger lib component

### CODE REVIEW

**Review by chipp972 — APPROVED** [review]

File: packages/manager-components/src/components/templates/base/base.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+              </OsdsChip>
+            </td>
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  </OsdsTable>
+);
+
+export const ListingTemplateProps = {
```
Comment (chipp972) [inline packages/manager-components/src/components/templates/base/base.stories.tsx:95]:
```suggestion
export const listingTemplateProps = {
```

File: packages/manager-components/src/components/templates/base/base.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  children: <Tabs />,
+  subtitle: '',
+  backLinkLabel: '',
+  onClickReturn: () => {
+    console.log('back link click');
+  },
+  subdescription: '',
+};
+
+const DashboardTemplateProps = {
```
Comment (chipp972) [inline packages/manager-components/src/components/templates/base/base.stories.tsx:129]:
```suggestion
const dashboardTemplateProps = {
```

File: packages/manager-components/src/components/templates/base/base.component.tsx [hunk 3]
```diff
@@ -2,41 +2,46 @@ import React from 'react';
...
   header?: HeadersProps;
   message?: React.ReactElement;
   description?: string;
   subtitle?: string;
   subdescription?: string;
   backLinkLabel?: string;
+  hrefPrevious?: string;
   tabs?: React.ReactElement;
   onClickReturn?: () => void;
+  children?: React.ReactElement | React.ReactElement[];
```
Comment (chipp972) [inline packages/manager-components/src/components/templates/base/base.component.tsx:18]:
```suggestion
  children?: React.ReactNode;
```

Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/src/components/templates/base/base.component.tsx:18]: 👍

File: packages/manager-components/src/components/templates/base/base.stories.tsx (hunk 2 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/src/components/templates/base/base.stories.tsx:129]: 👍

File: packages/manager-components/src/components/templates/base/base.stories.tsx (hunk 1 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/src/components/templates/base/base.stories.tsx:95]: 👍

**Review by chipp972 — DISMISSED** [review]

**Review by darsene — APPROVED** [review]

File: packages/manager-components/src/components/templates/base/base.component.tsx [hunk 4]
```diff
@@ -46,6 +51,6 @@ export const DashboardLayout = ({
       <Description className="mb-8">{subdescription}</Description>
     )}
     <div className="mb-6">{tabs}</div>
-    <div>{content}</div>
+    <div>{children}</div>
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/src/components/templates/base/base.component.tsx:54]: 👍

File: packages/manager-components/src/components/templates/base/base.component.tsx [hunk 5]
```diff
@@ -2,41 +2,46 @@ import React from 'react';
 import { Headers, HeadersProps } from '../../content';
 import { Description, LinkType, Links, Subtitle } from '../../typography';
 import { PageLayout } from '../layout/layout.component';
+import { OdsHTMLAnchorElementTarget } from '@ovhcloud/ods-common-core';
 
-export interface DashboardLayoutProps {
+export interface BaseLayoutProps {
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/src/components/templates/base/base.component.tsx]: 👍

File: packages/manager-components/src/components/templates/base/base.component.tsx [hunk 6]
```diff
@@ -2,41 +2,45 @@ import React from 'react';
...
 
-export interface DashboardLayoutProps {
+export type BaseLayoutProps = React.PropsWithChildren<{
   breadcrumb?: React.ReactElement;
-  content?: React.ReactElement;
   header?: HeadersProps;
   message?: React.ReactElement;
   description?: string;
   subtitle?: string;
   subdescription?: string;
```
Comment (anooparveti) [inline packages/manager-components/src/components/templates/base/base.component.tsx:13]: **question:** Can we change this to `subDescription`?

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx [hunk 7]
```diff
@@ -76,6 +79,27 @@ const Dashboard: React.FC<DashboardLayoutProps> = ({ tabs, rancher }) => {
...
+      setHasTaskPending(true);
+    }
+  }, [currentTasks]);
+
+  useEffect(() => {
+    if (hasTaskPending && currentTasks.length === 0) {
+      setIsPendingUpdate(false);
+      setHasTaskPending(false);
+    }
+  }, [currentTasks]);
```
Comment (anooparveti) [inline packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx:101]: **question:** Why is this change included in `BaseLayout` changes?

Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx:101]: Because I create an attribute DashboardMessages attributes to pass it below the tabs section

File: packages/manager-components/src/components/templates/base/base.component.tsx (hunk 6 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager-components/src/components/templates/base/base.component.tsx:13]: 👍

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail.tsx [hunk 8]
```diff
@@ -97,7 +97,7 @@ const RancherDetail = ({
     !updateSoftwareResponseType;
 
   return (
-    <div className="max-w-4xl">
+    <div>
```
[the remark this answers is not in the document]
Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/RancherDetail.tsx]: Finally I will update the rancher app in an other PR

File: packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx (hunk 7 above)
Comment (aboungnaseng-ovhcloud) [reply packages/manager/apps/pci-rancher/src/components/layout-helpers/Dashboard/Dashboard.tsx]: I will update rancher in an other PR to merge the component more easily

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]
